controlplane/kubeadm/controllers/scale.go

```diff
@@ -135,5 +135,8 @@ func selectMachineForScaleDown(controlPlane *internal.ControlPlane, outdatedMach
 	if outdatedMachines.Len() > 0 {
 		machines = outdatedMachines
 	}
+	if withDeleteAnnotation := controlPlane.MachineWithDeleteAnnotation(machines); withDeleteAnnotation != nil {
+		return withDeleteAnnotation, nil
+	}
```

Comment:
Let's move this at the top of this function, on line 134 given that the logic here effectively ignores any other criteria to pick a Machine to scale down, and always prefers the Machine that the user selected.

Reply:
If we remove the .Oldest call inside `MachineWithDeleteAnnotation`, we can return the whole list from that function and overwrite the `machines` slice here. We should also move back this block after the `outdatedMachines` one, so it overwrites the list.

Then we can return normally